src/strtree.c

```diff
@@ -16,6 +16,39 @@
 #include "pygeom.h"
 #include "kvec.h"
 
+/* GEOS function that takes a prepared geometry and a regular geometry
+ * and returns bool value */
+
+typedef char FuncGEOS_YpY_b(void *context, const GEOSPreparedGeometry *a,
+                            const GEOSGeometry *b);
+
+
+/* Copy values from arr to a new numpy integer array.
+ * The order of values from arr is inverted, because arr is created by pushing
+ * values onto the end. */
+
+static PyArrayObject *copy_kvec_to_npy(npy_intp_vec *arr)
+{
+    npy_intp i;
+    npy_intp size = kv_size(*arr);
+    npy_intp *ptr;
+
+    npy_intp dims[1] = {size};
+    // the following raises a compiler warning based on how the macro is defined
+    // in numpy.  There doesn't appear to be anything we can do to avoid it.
+    PyArrayObject *result = (PyArrayObject *) PyArray_SimpleNew(1, dims, NPY_INTP);
+    if (result == NULL) {
+        return NULL;
+    }
+
+    // iterate from last value to first to get back original order
+    for (i = size - 1; i >= 0; i--) {
+        ptr = PyArray_GETPTR1(result, i);
+        *ptr = (npy_intp)kv_a(npy_intp, *arr, i);
```

Comment:
Is there a reason you use `kv_a` here instead of `kv_pop`? (I assume the use of `kv_pop` was also the reason to iterate backwards; I don't know if there was a performance reason to use kv_pop over kv_a)

Reply:
I did not test the performance of either variant.  I had assumed there might be allocations involved in `kv_pop` as you remove items from the vector, but looking more closely at the implementation I don't believe that happens.  I just wanted to access an element, not remove it from the vector.

I use `kv_A` below instead of `kv_a`, so I wasn't very consistent either.  

Looping backwards was probably a holdover from using `kv_pop`.

I'll standardize to `kv_pop` - the implementation is much simpler so it theoretically should be marginally faster.

Reply:
Changed my mind, went with `kv_A` in both cases.  It's function signature looks low complexity, so it should be as fast as `kv_pop`, and allows us to iterate forward in arrays.